src/Data/Text/Internal/Fusion/Types.hs

```diff
@@ -29,10 +29,11 @@ module Data.Text.Internal.Fusion.Types
     ) where
 
 import Data.Text.Internal.Fusion.Size
+import Data.Int (Int64)
 import Data.Word (Word8)
 
 -- | Specialised tuple for case conversion.
-data CC s = CC !s {-# UNPACK #-} !Char {-# UNPACK #-} !Char
+data CC s = CC !s {-# UNPACK #-} !Int64
```

Comment:
It might make sense to rename the CC constructor and offer a pattern synonym to avoid unnecessarily breaking downstream. I guess the Int64 is really just the two chars squashed into a Word64? I am specifically looking at https://github.com/brendanhay/text-manipulate/blob/main/lib/Data/Text/Manipulate/Internal/Fusion.hs#L166 which would have a hard time to adjust to the new changes. 

Please ignore if you think this is a non-issue.

Reply:
Anything under `Internal` should be fair game to be changed.

Reply:
> I guess the Int64 is really just the two chars squashed?

Not quite, it's up to three chars squashed. 

I'm reluctant to calcify for eternity a pattern synonym here. This is a very much internal part with a non-descriptive name, which is not really intended for external usage. Important part is that the type has changed, so `text-manipulate` won't be silently broken.